plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d

```diff
@@ -564,23 +564,37 @@ struct MetaSpan {
         noCoverage,
     }
 
+    static struct MutationLength {
+        ulong length;
+        MutationId id;
+        Mutation.Status status = Mutation.Status.killed;
+
+        this(ulong length, MutationId id, Mutation.Status status) {
+            this.length = length;
+            this.id = id;
+            this.status = status;
+        }
+    }
+
     StatusColor status;
     string onClick;
+    MutationLength clickPrio;
 
     this(const(FileMutant)[] muts) {
         immutable click_fmt2 = "ui_set_mut(%s)";
         status = StatusColor.none;
-
+        if (muts.length != 0) {
+            clickPrio = MutationLength(muts[0].txt.mutation.length, muts[0].id, muts[0].mut.status);
+        }
         foreach (ref const m; muts) {
             status = pickColor(m, status);
-            if (onClick.length == 0 && m.mut.status == Mutation.Status.alive) {
-                onClick = format(click_fmt2, m.id.get);
+            if (m.mut.status == Mutation.Status.alive && clickPrio.status != Mutation.Status.alive) {
+                clickPrio = MutationLength(m.txt.mutation.length, m.id, m.mut.status);
+            } else if (m.txt.mutation.length < clickPrio.length && m.mut.status == clickPrio.status) {
```

Comment:
questions.
1. the desired behavior is that the smallest alive is shown. i think the impl handle that.
2. if there are no alive then the desired is that the smallest mutant is shown irregardless of status.
- i do not think the impl that that case?

Reply:
ok so just removing the last statement in the elseif statement would fix that right?

Reply:
the algo was wrong, testing helped to build it better. Thank you!